### DISCUSSION TITLE
Is there any commands or options for starting lima with boot?

### DISCUSSION THREAD

**Discussion by JJHWAN** [lima-vm/lima#1027] [Q&A]
Thanks for the answer with My previous Q&A.

Next Now I'm trying to set Lima VM started with my Mac OS boot

Is there options or commands for setting auto reboot with system or user login?

I have tried with launchd (launchctl) Agents & Daemons,

but with agent, VM disappears after it's starting  process is done
(setting KeepAlive flag makes loop with starting process)

and with daemon, error [time="..." level=fatal msg="must not run as the root"] comes out.

Is there a Lima VM user using Mac with auto reboot of VM?

**Comment by Turbulentarius** [comment]
I think you need to use the "Automator.app", and then run a .sh bash script on login, the contents being similar to:

`limactl start [name-of-vm]`

**Comment by mhio** [comment]
The following plist via `launchctl load ~/Library/LaunchAgents/com.me.vm.lima.debian.plist` worked for me
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>KeepAlive</key>
    <true/>
    <key>Label</key>
    <string>com.me.vm.lima.debian</string>
    <key>ProgramArguments</key>
    <array>
      <string>/usr/local/bin/limactl</string>
      <string>hostagent</string>
      <string>--pidfile</string>
      <string>/Users/me/.lima/debian/ha.pid</string>
      <string>--socket</string>
      <string>/Users/me/.lima/debian/ha.sock</string>
      <string>debian</string>
    </array>
    <key>RunAtLoad</key>
    <true/>
    <key>UserName</key>
    <string>me</string>
    <key>WorkingDirectory</key>
    <string>/Users/me</string>
    <key>StandardErrorPath</key>
    <string>/Users/me/.lima/debian/launchd.log</string>
    <key>StandardOutPath</key>
    <string>/Users/me/.lima/debian/launchd.log</string>
  </dict>
</plist>
```

The VM doesn't stop very cleanly though, I think `limactl hostagent` immediately exits on a SIGTERM so you need to `limactl stop vm` and possibly `limactl disk unlock` to get a VM to start up again.
